cfme/automate/dialog_element.py

```diff
@@ -0,0 +1,196 @@
+            format(self.parent.parent.parent.label))
+        view.flash.assert_no_error()
+        return self.instantiate(element_data=element_data)
+
+    def element_type(self, view, each_element):
+        """ Method to add element type.Depending on their type the subfields varies.
+
+        Args:
+            each_element: subfields depending on element type.
+        """
+        choose_type = each_element.get("choose_type")
+        dynamic_chkbox = each_element.get("dynamic_chkbox")
+        element_type = ['Drop Down List', 'Radio Button']
+        if choose_type in element_type:
+            if not dynamic_chkbox:
+                row = view.entry_table.row(Value='<New Entry>')
+                row.click()
+                view.fill({'entry_value': "Yes",
+                           'entry_description': "entry_desc"})
+                view.add_entry_button.click()
+            else:
+                node1 = "InspectMe"
+                view.fill({'field_entry_point': 'b'})
+                view.bt_tree.click_path("Datastore", "new_domain", "System", "Request", node1)
+                view.apply_btn.click()
+                view.fill({'field_show_refresh_button': True})
+        if choose_type == "Text Area Box":
+            view.fill({'text_area': 'Default text'})
+
+
+class Element(Navigatable):
+    """A class representing one Element of a dialog."""
+    def __init__(self, collection, element_data):
+        Navigatable.__init__(self, appliance=collection.appliance)
+        self.collection = collection
+        self.element_data = element_data
+
+    @property
+    def parent(self):
+        return self.collection.parent
+
+    @property
+    def tree_path(self):
+        return self.collection.tree_path
+
+    @property
+    def box(self):
+        return self.parent.box
+
+    def element_loc(self, element_data):
+        return self.browser.element('//div[@class="panel-heading"]'
+            '[contains(normalize-space(.), {})]/..'.format(quote(element_data)))
+
+    def reorder_elements(self, add_element, second_element, element_data):
+        """Method to add element and interchange element positions.
+           This method updates a dialog and adds a second element.The position
+           of two elements are then interchanged to test for error.
+
+        Args:
+            add_element - flag if second element needs to be added.
+        """
+        view = navigate_to(self, 'Edit')
+        view.element_tree.click_path(self.parent.parent.parent.label,
+            self.parent.parent.tab_label,
+            self.parent.box_label)
+        # Add a new element and then interchange position (BZ-1238721)
+        if add_element:
+            view.plus_btn.item_select("Add a new Element to this Box")
+            view.fill(second_element)
+            view.element_tree.click_path(self.parent.parent.parent.label,
+                self.parent.parent.tab_label,
+                self.parent.box_label)
```

Comment:
Why we can't add the element using the collection? This is mixing of responsibilities between colleciton and an element.